### PR TITLE
Add text filter to convert absolute urls on the same dept site to relative ones

### PR DESCRIPTION
**Pull request by johangant**
Differs from the RelToAbsUrl filter as this goes the other direction + is used for converting links to absolute values on top-level sites like NIGOV.

### CODE REVIEW

File: web/modules/custom/dept_core/modules/dept_content_processors/src/Plugin/Filter/AbsToRelUrlsFilter.php [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+   * @param string $plugin_id
+   *   The plugin ID for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \Drupal\dept_core\DepartmentManager $department_manager
+   *   The department manager object.
+   *
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, DepartmentManager $department_manager) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
```
Comment (omahm) [inline web/modules/custom/dept_core/modules/dept_content_processors/src/Plugin/Filter/AbsToRelUrlsFilter.php:48]: I'd look at using the $instance pattern which doesn't override the base constructor.

File: web/modules/custom/dept_core/modules/dept_content_processors/src/Plugin/Filter/AbsToRelUrlsFilter.php [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        if (str_starts_with($href, 'http')) {
+          // Look at the hostname, if it matches our current domain
+          // then remove the hostname from the attribute and update the
+          // dom element.
+          $url_portions = parse_url($href);
+          $host = $url_portions['host'];
+
+          if (!empty($url_portions['path']) && $this->hostnameMatchesKnownDepartment($host, $dept->id())) {
+            $new_href = $url_portions['path'];
+            if (!empty($url_portions['query'])) {
```
Comment (omahm) [inline web/modules/custom/dept_core/modules/dept_content_processors/src/Plugin/Filter/AbsToRelUrlsFilter.php:91]: I'd also check for the hashbang

**Review by omahm — CHANGES_REQUESTED** [review]
Looks good, just a couple of comments

File: web/modules/custom/dept_core/modules/dept_content_processors/src/Plugin/Filter/AbsToRelUrlsFilter.php (hunk 1 above)
Comment (johangant) [reply web/modules/custom/dept_core/modules/dept_content_processors/src/Plugin/Filter/AbsToRelUrlsFilter.php:47]: Having another attempt at this today but it's functional with just using the dept manager service, so would see the context injection as an enhancement.

**Review by omahm — APPROVED** [review]
LGTM

**Pull request merged**